### PR TITLE
github workflow: run arm64 tests on every push

### LINKED ISSUE #16905: Run `arm64` integration tests for every pull request rather than on nightly schedule
### What would you like to be added?

Recently in https://github.com/etcd-io/etcd/pull/16801 we introduced on demand github actions runners for the `arm64` platform.

Having on demand runner infrastructure in place means we should now have enough capacity to begin running `arm64` tests for every pull request.

We should update our `.github/workflows/tests-arm64-nightly.yaml` workflow to switch from scheduled nightly runs to instead running `on: [push, pull_request]`: https://github.com/etcd-io/etcd/blob/main/.github/workflows/tests-arm64-nightly.yaml#L7-L8

### Why is this needed?

Gain greater confidence and samples of integration tests running against `arm64` cpu architecture.

Hopefully detect issues before pull requests are merged to `main`.

### PR DESCRIPTION
**Pull request by ivanvc**
Use a template to consolidate arm64 and amd64 test workflows. Enable running armd64 tests on every push and pull request.

I think this change will also need to be applied to the release-35 branch, as it will no longer run nightly on that branch. If someone can confirm, I can open a PR targeting that branch.

Fixes #16905.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by jmhbnz** [comment]
Thanks @ivanvc.  @ahrtr can you please approve workflows for this first time contributor?

**Review by serathius — APPROVED** [review]

File: .github/workflows/tests-arm64-nightly.yaml [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-  # schedules always run against the main branch, hence we have to create separate jobs
-  # with individual checkout actions for each of the active release branches
-  schedule:
-    - cron: '30 2 * * *' # runs daily at 2:30 am.
-jobs:
-  main-arm64:
-    uses: ./.github/workflows/tests-arm64-template.yaml
-    with:
-      etcdBranch: main
-  release-35-arm64:
```
Comment (ahrtr) [inline .github/workflows/tests-arm64-nightly.yaml:14]: Are you going to take care of release-3.5 in a separate PR?

Comment (ivanvc) [reply .github/workflows/tests-arm64-nightly.yaml:14]: Yes, I'll create a new PR targeting the release-35 branch.

**Review by ahrtr — APPROVED** [review]
lgtm

**Pull request merged**